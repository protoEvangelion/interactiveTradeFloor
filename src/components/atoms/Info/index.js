import React, { PropTypes } from 'react'
import styled from 'styled-components'

const BoothNum = styled.h3`
  color: black;
  font-size: 1em;
  margin: 0;
  text-align: center;
  padding-top: 4px;
`

const BoothCo = styled.p`
  color: black;
  font-size: .68em;
  margin: 0;
  text-align: center;
  padding: 4px 2px 0 2px;
  overflow-wrap: break-word;
  line-height: 1.3em;
`

const Info = ({ num, co, status, tip }) => {
  return (
    <div data-tip data-for={tip} >
      <BoothNum>{ num }</BoothNum>
      <BoothCo>{ status === 'open' ? status : co }</BoothCo>
    </div>
  )
}

Info.propTypes = {
  num: PropTypes.number,
  co: PropTypes.string,
  status: PropTypes.string,
  tip: PropTypes.string.isRequired,
}

export default Info
