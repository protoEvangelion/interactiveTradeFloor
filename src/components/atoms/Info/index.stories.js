import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Info from '.'

storiesOf('Info', module)
  .add('default', () => (
    <Info>Hello</Info>
  ))
  .add('reverse', () => (
    <Info reverse>Hello</Info>
  ))
