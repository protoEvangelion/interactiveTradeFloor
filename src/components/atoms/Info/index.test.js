import React from 'react'
import renderer from 'react-test-renderer'
import Info from '.'

describe('<Info /> snapshot', () => {
	it('renders with open status', () => {
		const tree = renderer.create(<Info tip="Hello" status="open" />).toJSON()
		expect(tree).toMatchSnapshot()
	})
	it('renders with holding status', () => {
		const tree = renderer.create(<Info tip="Hello" status="Holding" />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
