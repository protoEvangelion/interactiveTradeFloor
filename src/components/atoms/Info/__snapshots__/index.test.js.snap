// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> snapshot renders with holding status 1`] = `
<div
  data-for="Hello"
  data-tip={true}
>
  <h3
    className="kzkpta"
  />
  <p
    className="fAiHFE"
  />
</div>
`;

exports[`<Info /> snapshot renders with open status 1`] = `
<div
  data-for="Hello"
  data-tip={true}
>
  <h3
    className="kzkpta"
  />
  <p
    className="fAiHFE"
  >
    open
  </p>
</div>
`;
